app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -16,19 +16,179 @@
+                    message = "Unknown error";
+                }
+                new AlertDialog.Builder(this).setMessage(message)
+                        .setNeutralButton(android.R.string.ok, null).show();
+            }
+        }
+    }
+
+    private void onDisconnected() {
+        isActiveConnection = true;
+        setContentView(R.layout.activity_sign_in);
+        findViewById(R.id.sign_in_button).setOnClickListener(this);
+    }
+
+    private void onConnected(GoogleSignInAccount googleSignInAccount) {
+        Log.d(TAG, "Into on connected method");
+        if (savedSignedInAccount != googleSignInAccount) {
+            savedSignedInAccount = googleSignInAccount;
+            // Get the playerId from the PlayersClient
+            PlayersClient playersClient = Games.getPlayersClient(this, googleSignInAccount);
+            playersClient.getCurrentPlayer()
+                    .addOnSuccessListener(new OnSuccessListener<Player>() {
+                        @Override
+                        public void onSuccess(Player player) {
+                            mPlayerId = player.getPlayerId();
+                            Log.d(TAG, "Player name : " + player.getDisplayName());
+                            switchToGameScreen();
+                            if (isActiveConnection)
+                                isActiveConnection = false;
+                        }
+                    })
+                    .addOnFailureListener(createFailureListener("There was a problem getting the player id!"));
+        }
+    }
+
+    private OnFailureListener createFailureListener(final String string) {
+        return new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception e) {
+                Log.e(TAG, string);
+                onDisconnected();
+            }
+        };
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view.getId() == R.id.sign_in_button) {
+            startSignInIntent();
+        } else if (view.getId() == R.id.sign_out_button) {
+            signOut();
+        }
+    }
+
+    private void signOut() {
+        GoogleSignInClient signInClient = GoogleSignIn.getClient(this,
+                GoogleSignInOptions.DEFAULT_GAMES_SIGN_IN);
+        signInClient.signOut().addOnCompleteListener(this,
+                new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        // at this point, the user is signed out.
+                        onDisconnected();
+                    }
+                });
+    }
+
+    void switchToGameScreen() {
         LevelManager.init(this);
         getSupportFragmentManager().beginTransaction().replace(R.id.activity_wrapper, new StartFragment()).commit();
         setContentView(R.layout.main_activity);
+        findViewById(R.id.sign_out_button).setOnClickListener(this);
```

Comment:
this doesn't seem the right place to have this line. My guess it should be in theFragment ?

Reply:
Done